### PR TITLE
Migrate readme yaml

### PR DESCRIPTION
**Pull request by vadim-gleyf**
## what
* Add `README.yaml`

## why
* Standardize README

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+github_repo: cloudposse/github-status-updater
+
+# Badges to display
+badges:
+  - name: "Build Status"
+    image: "https://travis-ci.org/cloudposse/github-status-updater.svg?branch=master"
+    url: "https://travis-ci.org/cloudposse/github-status-updater"
+  - name: "Latest Release"
+    image: "https://img.shields.io/github/release/cloudposse/github-status-updater.svg"
+    url: "https://travis-ci.org/cloudposse/github-status-updater/releases"
```
Comment (osterman) [inline README.yaml]: Wrong URL

File: README.yaml [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+  ### run in a Docker container with ENV vars declared in a file
+  [run_docker_with_env_vars_file.sh](examples/run_docker_with_env_vars_file.sh)
+
+  ```sh
+  docker run -i --rm --env-file ./example.env github-status-updater
+  ```
+
+
+  ###
```
Comment (osterman) [inline README.yaml]: Why do we have these `###` here

File: README.yaml [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+          -e GITHUB_STATE=success \
+          -e GITHUB_CONTEXT="my-ci" \
+          -e GITHUB_DESCRIPTION="Commit status with target URL" \
+          -e GITHUB_TARGET_URL="https://my-ci.com/build/1" \
+          github-status-updater
+  ```
+
+
+
+  ### run in a Docker container with local ENV vars propagated into the container's environment
```
Comment (osterman) [inline README.yaml]: I want to take this opportunity to clean this up. I don't like titles that are long & drawn out sentences. 

Change this to:
```
### Run with Docker

Run the `github-status-updater` in a Docker container with local ENV vars propagated into the container's environment.
```

Comment (osterman) [reply README.yaml]: Please take a stab at doing the other title sections this way.

File: README.yaml [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+          -e GITHUB_STATE \
+          -e GITHUB_CONTEXT \
+          -e GITHUB_DESCRIPTION \
+          -e GITHUB_TARGET_URL \
+          github-status-updater
+  ```
+
+
+
+  ### run in a Docker container with ENV vars declared in a file
```
Comment (osterman) [inline README.yaml]: Fix chapter title

File: README.yaml [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+  export GITHUB_TARGET_URL="https://my-ci.com/build/1"
+
+  ./dist/bin/github-status-updater
+  ```
+
+
+  After the above command is executed, the commit status will be updated to `success` with the target URL `https://my-ci.com/build/1` (the green check mark in the image below)
+
+  ![GitHub Commit Status](images/github-commit-status.png)
+  ###
```
Comment (osterman) [inline README.yaml:101]: Why is this here?

Comment (osterman) [reply README.yaml:101]: @aknysh can maybe elaborate

File: README.yaml [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+  | repo                  | GITHUB_REPO         | Github repository name                                                         |
+  | ref                   | GITHUB_REF          | Commit SHA, branch name or tag                                                 |
+  | state                 | GITHUB_STATE        | Commit state. Possible values are `pending`, `success`, `error` or `failure`   |
+  | context               | GITHUB_CONTEXT      | Status label. Could be the name of a CI environment (_e.g._ `my-ci`)           |
+  | description           | GITHUB_DESCRIPTION  | Short high level summary of the status                                         |
+  | url                   | GITHUB_TARGET_URL   | URL of the page representing the status                                        |
+
+
+
+  ### build the Go program locally
```
Comment (osterman) [inline README.yaml]: Properly capitalize all chapter titles

File: README.yaml [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+          -ref XXXXXXXXXXXXXXX \
+          -state success \
+          -context "my-ci" \
+          -description "Commit status with target URL" \
+          -url "https://my-ci.com/build/1"
+  ```
+
+
+
+  ### build the Docker image
```
Comment (osterman) [inline README.yaml]: Capitalize title

File: README.yaml [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+  ```
+
+
+  After the above command is executed, the commit status will be updated to `success` with the target URL `https://my-ci.com/build/1` (the green check mark in the image below)
+
+  ![GitHub Commit Status](images/github-commit-status.png)
+  ###
+
+
+  ### run locally with command-line arguments
```
Comment (osterman) [inline README.yaml]: Capitalize

File: README.yaml [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+
+  ![GitHub Status Checks](images/codefresh-deployment-status-pending.png)
+  ###
+
+  ![GitHub Status Checks](images/codefresh-deployment-status-success.png)
+  ###
+
+
+
+  ## References
```
Comment (osterman) [inline README.yaml]: Let's add a `references` type to the `build-harness` so we can write

```
references:
  - https://github.com/google/go-github
  - https://godoc.org/github.com/google/go-github/github
```

File: README.yaml (hunk 5 above)
Comment (vadim-gleyf) [reply README.yaml:101]: @osterman I think it was planned. 
`After the above command is executed, the commit status will be updated to success with the target URL https://my-ci.com/build/1 (the green check mark in the image below)`

**Comment by mkryvitski** [comment]
Ok

**Comment by osterman** [comment]
@solairerove please rebase

**Comment by solairerove** [comment]
great changes
